File: CPP/Processor.cpp
class Processor
{
    private:
        /* data */
        string name;
        int price;

    public:
        Processor(/* args */) { }
        Processor(string n, int p) { 
            this->name = n;
            this->price = p;
        }

        void setName(string n) { this->name = n; }
        void setprice(int p) { this->price = p; }

        string getName() { return this->name; }
        int getPrice() { return this->price; }

        void printDetProcessor(){
            cout << "----------------------" << endl;
            cout << "Processor Name : " << this->name << endl;
            cout << "Price          : " << this->price << endl;
        }

        ~Processor(){ }
};

File: PHP/Processor.php
<?php

class Processor{
    private $name;
    private $price;

    public function __construct($n, $p){
        $this->name = $n;
        $this->price = $p;
    }

    public function setName($n){
        $this->name = $n;
    }
    public function getName(){
        return $this->name;
    }

    public function setPrice($p){
        $this->price = $p;
    }
    public function getPrice(){
        return $this->price;
    }

    public function printProcessor(){
        echo "---------------------- <br>";
        echo "Processor Name : ". $this->name ."<br>";
        echo "Price          : ". $this->price ."<br>";
    }

    public function __destruct(){ }
}

?>